### PR TITLE
[Feature] create and drop hive database in the hive catalog

### PR DESCRIPTION
**Pull request by stephen-shelby**
Fixes #issue
Support create and drop hive database.
- create database hive_db; // under hive catalog
- create database hive_catalog.hive_db properties ("location"="s3://bucket/hive_db")
- drop database ut1; // only empty database can be dropped. The db location dir will be deleted only if there is no directory or file under the db location

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 55 / 75 (73.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadataFactory.java|0|2|00.00%|[44, 46]|
|:large_blue_circle:|com/starrocks/connector/hive/CachingHiveMetastore.java|1|8|12.50%|[190, 191, 192, 197, 527, 530, 531]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetaClient.java|2|5|40.00%|[188, 189, 190]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadataFactory.java|2|4|50.00%|[60, 67]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastore.java|5|6|83.33%|[75]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreOperations.java|27|31|87.10%|[55, 82, 83, 84]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|9|10|90.00%|[89]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 67 / 85 (78.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetaClient.java|0|7|00.00%|[182, 184, 185, 187, 190, 191, 193]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadataFactory.java|0|2|00.00%|[44, 46]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|7|9|77.78%|[92, 121]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreOperations.java|30|37|81.08%|[70, 71, 75, 79, 94, 104, 107]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastore.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadataFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/CachingHiveMetastore.java|9|9|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/CachingHiveMetastore.java [hunk 1]
```diff
@@ -180,6 +180,24 @@ private List<String> loadAllDatabaseNames() {
         return metastore.getAllDatabaseNames();
     }
 
+    @Override
+    public void createDb(String dbName, Map<String, String> properties) {
+        try {
+            metastore.createDb(dbName, properties);
+        } finally {
+            invalidateDatabase(dbName);
```
[the remark this answers is not in the document]
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/CachingHiveMetastore.java:188]: `getDb` will cache the database entry.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 67 / 85 (78.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetaClient.java|0|7|00.00%|[182, 184, 185, 187, 190, 191, 193]|
|:large_blue_circle:|com/starrocks/connector/delta/DeltaLakeMetadataFactory.java|0|2|00.00%|[44, 46]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadata.java|7|9|77.78%|[92, 121]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreOperations.java|30|37|81.08%|[70, 71, 75, 79, 94, 104, 107]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastore.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetadataFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadataFactory.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/CachingHiveMetastore.java|9|9|100.00%|[]|

**Review by DorianZheng — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**